torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -120,44 +123,161 @@ def can_auto_functionalize(op: OperatorBase) -> bool:
     return True
 
 
+@dataclass
+class ViewInfo:
+    base_index: int
+    size: Any = None
+    stride: Any = None
+    storage_offset: Any = None
+    is_alias_to_base: bool = False
+
+
+def serialize_views_meta(
```

Comment:
serialize_view_meta and deserialize_view_meta are not actually inverses of each other, right? Can we make them inverses? (only if it helps make things clearer). So, I'd expect that we actually construct ViewInfo and then pass them to serialize_views_meta